plugin-cgroupv2/src/k8s/pod.rs

```diff
@@ -0,0 +1,929 @@
+            log::error!("Error parsing JSON: {}", err);
+            Value::Null
+        }
+    };
+
+    let mut pod_infos_by_uid = HashMap::new();
+
+    // let's check if the items' part contain pods to look at
+    if let Some(items) = data.get("items") {
+        // If the node was not found i.e. no item in the response, we call the API again with all nodes
+        let size = items.as_array().map(|a| a.len()).unwrap_or(0);
+        if size == 0 && selector {
+            // Ask again the api, with all nodes
+            let Ok(response) = client.get(api_url_root).send().await else {
+                return Ok(HashMap::new());
+            };
+
+            data = match response.json().await {
+                Ok(value) => value,
+                Err(err) => {
+                    log::error!("Error parsing JSON: {}", err);
+                    Value::Null
+                }
+            }
+        } else {
+            log::debug!("Data is empty or not available.");
+        }
+    }
+
+    if let Some(items) = data.get("items") {
+        for item in items.as_array().unwrap_or(&vec![]) {
+            let metadata = item.get("metadata").unwrap_or(&Value::Null);
+            let spec = item.get("spec").unwrap_or(&Value::Null);
+            let annotations = metadata.get("annotations").unwrap_or(&Value::Null);
+            let mut config_hash = annotations
+                .get("kubernetes.io/config.hash")
+                .and_then(|v| v.as_str())
+                .unwrap_or("");
+
+            if config_hash.is_empty() {
+                match metadata {
+                    Value::Null => {
+                        continue;
+                    }
+                    _ => {
+                        config_hash = metadata.get("uid").and_then(|v| v.as_str()).unwrap_or("");
+                    }
+                }
+            }
+
+            let pod_name = metadata.get("name").and_then(|v| v.as_str()).unwrap_or("");
+            let pod_namespace = metadata.get("namespace").and_then(|v| v.as_str()).unwrap_or("");
+            let pod_node = spec.get("nodeName").and_then(|v| v.as_str()).unwrap_or("");
+
+            pod_infos_by_uid.entry(String::from(config_hash)).or_insert(PodInfos {
+                name: pod_name.to_owned(),
+                namespace: pod_namespace.to_owned(),
+                node: pod_node.to_owned(),
+            });
+        }
+    } else {
+        log::debug!("No items part found in the JSON response.");
+    }
+
+    Ok(pod_infos_by_uid)
+}
+
+/// Reads files in a filesystem to associate a cgroup of a pod uid to a kubernetes pod name
+pub async fn get_pod_infos(uid: &str, node: &str, kubernetes_api_url: &str, token: &Token) -> anyhow::Result<PodInfos> {
+    let new_uid = uid.replace('_', "-");
+    let token = match token.get_value().await {
+        Ok(token) => token,
+        Err(e) => {
+            log::error!("Could not retrieve the token, got: {e}");
+            return Ok(PodInfos::empty());
```

Comment:
The error handling of this function is inconsistent: it returns a Result but 99% of the calls that can fail returns Ok(PodInfos::empty). The error should be propagated instead, and the caller should decide to use empty or not.